src/main/scala/RocketChip.scala

```diff
@@ -11,16 +11,26 @@ import rocket.Util._
 
 /** Top-level parameters of RocketChip, values set in e.g. PublicConfigs.scala */
 
+/** Options for memory bus interface */
+object BusType {
+  sealed trait EnumVal
+  case object AXI extends EnumVal
+  case object AHB extends EnumVal
+  val busTypes = Seq(AXI, AHB)
+}
+
 /** Number of memory channels */
 case object NMemoryChannels extends Field[Int]
+case object TMemoryChannels extends Field[BusType.EnumVal]
 /** Number of banks per memory channel */
 case object NBanksPerMemoryChannel extends Field[Int]
 /** Least significant bit of address used for bank partitioning */
 case object BankIdLSB extends Field[Int]
 /** Number of outstanding memory requests */
 case object NOutstandingMemReqsPerChannel extends Field[Int]
 /** Number of exteral MMIO ports */
-case object NExtMMIOChannels extends Field[Int]
+case object NExtMMIOAXIChannels extends Field[Int]
+case object NExtMMIOAHBChannels extends Field[Int]
```

Comment:
It's probably a safe assumption in the memory case that the channels are homogeneous, but not for MMIO, where having both AXI and AHB bridges is conceivable.